src/domain/ids/Uuidv4.ts

```diff
@@ -0,0 +1,16 @@
+import { v4 as uuidv4 } from 'uuid';
+
+export class Uuidv4 {
+  value: string;
+
+  constructor(value: string, idType: string) {
+    if (value == null || value == '') {
+      throw new Error(`Invalid id format for: ${idType}`);
+    }
```

Comment:
Tu pourrais aussi le valider avec le meme module et `superstruct` si tu veux y aller au max: 
- https://www.npmjs.com/package/uuid#uuidvalidatestr
- https://github.com/ianstormtaylor/superstruct#usage

Reply:
I like that